### ISSUE TITLE
TrajOpt SCO Large Heap Allocation

### ISSUE THREAD

**Issue by Levi-Armstrong** [tesseract-robotics/trajopt#309]
Current when running the trajopt planning unit test leveraging heaptrack it was found to have allocated around 2GB of memory. This occurs in [solver_utils.cpp](https://github.com/tesseract-robotics/trajopt/blob/master/trajopt_sco/src/solver_utils.cpp) when space is reserved in the Eigen sparse matrix. Need to investigate if we are reserving the correct amount.

![image](https://user-images.githubusercontent.com/9803128/224752616-723ed184-08ce-4d6f-81ff-20ed83f70ba6.png)

**Comment by Levi-Armstrong** [comment]
I have been able to reduce the almost all of the memory allocation. I will have a PR very soon to address the issue. 
![image](https://user-images.githubusercontent.com/9803128/224779417-ff71b8d7-cb93-41fe-82cd-7e9c6d75faa2.png)

**Comment by Levi-Armstrong** [comment]
This also has significant performance increase 50X

Before:
```
---------------------------------------------------------------------------------------------------
Benchmark                                                         Time             CPU   Iterations
---------------------------------------------------------------------------------------------------
BM_TRAJOPT_SIMPLE_COLLISION_SOLVE/real_time                    3204 us         3204 us          217
BM_TRAJOPT_PLANNING_SOLVE/min_time:6.000/real_time            46973 ms        46970 ms            1
```

After:
```
```

**Comment by Levi-Armstrong** [comment]
Addressed by PR #310
